src/lib/y2partitioner/widgets/device_table_entry.rb

```diff
@@ -163,6 +167,7 @@ def children(device)
             []
           end
         end
+        # rubocop:enable all
```

Comment:
NP: Instead of disabling Rubocop, I'd extract the logic to small methods. Something like

```suggestion
        def children(device)
          return btrfs_subvolumes(device) if nesting_btrfs_subvolumes?(device)
          return lvm_lvs(device) if nesting_lvm_lvs?(device)
          
          partitions
       end
```

Reply:
This will not help, as you'd end up with around the same number of branches and get the complexity warnings again.

You could hide the logic in the individual classes but I'm not sure there's a point.

Reply:
If at all, then maybe something like

```
def children(device)
  btrfs_subvolumes(device) + lvm_lvs(device) + partitions(device)
end
```

Reply:
I tested doing 

```ruby
        def children(device)
          return btrfs_subvolumes(device) if nesting_btrfs_subvolumes?(device)
          return lvm_lvs(device) if nesting_lvm_lvs?(device)

          partitions(device)
        end

        def nesting_lvm_lvs?(device)
          device.is?(:lvm_vg) || device.is?(lvm_lv)
        end

        def lvm_lvs(device)
          if device.is?(:lvm_vg)
            # All logical volumes, including thin pools but excluding thin volumes
            device.all_lvm_lvs - device.thin_lvm_lvs
          elsif device.is?(:lvm_lv)
            # Logical volumes in a thin pool, if any; empty for normal logical volumes
            device.lvm_lvs
          end
        end

        def partitions(device)
          if device.respond_to?(:partitions)
            # Logical partitions are not included because they are nested within the extended one
            device.partitions.reject { |p| p.type.is?(:logical) }
          elsif device.is?(:partition) && device.type.is?(:extended)
            # All logical partitions
            device.children
          else
            []
          end
        end

        ...
```

and Rubocop didn't complain. 

Anyway, it was an NP comment.

Reply:
Aside note, 

> ```ruby
>           device.is?(:lvm_vg) || device.is?(:lvm_lv)
> ```

can be 

> ```ruby
>           device.is?(:lvm_vg, :lvm_lv)
> ```

Reply:
My point was why an extra 'if' if you could just return an empty list and simply add up. That's just asking for 'if' and 'list' logic to diverge and getting surprising results.

Reply:
> My point was why an extra 'if' if you could just return an empty list and simply add up. That's just asking for 'if' and 'list' logic to diverge and getting surprising results.

I overlooked your previous comment. Probably a browser cache problem. Fine with me.